### PR TITLE
Fix clippy warning

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

**Review by zbraniecki — DISMISSED** [review]

**Comment by sffc** [comment]
This is a duplicate of #921 but I will close #921 and let @Manishearth take it over.  Please investigate the additional clippy errors reported by CI.

**Comment by Manishearth** [comment]
Annoyingly CI only reports a single crate's clippy issue and moves on

I suspect this happened due to a compiler upgrade somewhere

File: tools/datagen/src/bin/datagen.rs [hunk 1]
```diff
@@ -322,10 +322,8 @@ fn get_blob_exporter(matches: &ArgMatches) -> anyhow::Result<BlobExporter<'stati
 
     let output_path: Option<PathBuf> = if matches.is_present("OUTPUT_TESTDATA") {
         Some(icu_testdata::paths::data_root().join("testdata.postcard"))
-    } else if let Some(v) = matches.value_of_os("OUTPUT") {
-        Some(PathBuf::from(v))
     } else {
-        None
+        matches.value_of_os("OUTPUT").map(PathBuf::from)
     };
```
Comment (sffc) [inline tools/datagen/src/bin/datagen.rs:327]: Nit (non-actionable): I find the previous code more readable, but I don't care enough to override clippy.

**Review by sffc — APPROVED** [review]

**Review by sffc — DISMISSED** [review]

**Comment by Manishearth** [comment]
Ugh we have more warnings under `--all-targets`

**Review by sffc — APPROVED** [review]

**Pull request merged**